cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -599,7 +510,10 @@ template <
                                     std::is_same<ParquetIntegerType, Int64Type>::value>>
 static Status DecimalIntegerTransfer(RecordReader* reader, MemoryPool* pool,
                                      const std::shared_ptr<DataType>& type, Datum* out) {
-  DCHECK_EQ(type->id(), ::arrow::Type::DECIMAL);
+  // Decimal256 isn't relevant here because this library never writes
```

Comment:
"This library" meaning parquet-cpp? But what do other implementations do?

Reply:
I added some clarification to this comment.  Let me know if it makes sense.